src/main/Config.h

```diff
@@ -258,6 +258,27 @@ class Config : public std::enable_shared_from_this<Config>
     // configuration) to delay emitting metadata by one ledger.
     bool EXPERIMENTAL_PRECAUTION_DELAY_META;
 
+    // A config parameter that when set uses the BucketList as the primary
+    // key-value store for LedgerEntry lookups
+    bool EXPERIMENTAL_BUCKET_KV_STORE;
```

Comment:
discussed offline a bit but following up for posterity: I think we need to make sure that when we switch to the bucketlist as a store, we should clean up any old state to avoid bugs and inconsistencies. This means that we need to drop the DB on restart if the experimental flag is one (so not just new-db), and also ensure that in the catchup path we don't actually apply buckets (so I think ApplyBucketsWork only needs to assumeState at that point)

Reply:
or actually, we would still need to apply buckets to rebuild offers, but we shouldn't be rebuilding any other tables

Reply:
With these changes, there are no errors and everything works as expected when switching the experimental flag from 0 to 1. However, if you switch the flag from 1 to 0, the app crashes with the error message `Got an exception: sqlite3_statement_backend::prepare: no such table:`. I don't expect many validators to switch the experimental flag off after turning it on, but should there be a better error message? For example, should we add a better error suggesting the user run `new-db`? Right now it just says to report the bug with the internal DB exception.